Analysis/Tasks/PWGHF/taskJpsi.cxx

```diff
@@ -61,19 +62,36 @@ struct TaskJpsi {
     registry.add("hDecLenXYErr", "2-prong candidates;decay length xy error (cm);entries", {HistType::kTH2F, {{100, 0., 0.01}, {(std::vector<double>)bins, "#it{p}_{T} (GeV/#it{c})"}}});
   }
 
-  Filter filterSelectCandidates = (aod::hf_selcandidate_jpsi::isSelJpsiToEE >= d_selectionFlagJpsi);
+  Filter filterSelectCandidates = (aod::hf_selcandidate_jpsi::isSelJpsiToEE >= d_selectionFlagJpsi || aod::hf_selcandidate_jpsi::isSelJpsiToMuMu >= d_selectionFlagJpsi);
 
   void process(soa::Filtered<soa::Join<aod::HfCandProng2, aod::HFSelJpsiCandidate>> const& candidates)
   {
+    int decaymode = DecayType::JpsiToEE;
+    if (d_modeJpsiToMuMu)
+      decaymode = DecayType::JpsiToMuMu;
+
     for (auto& candidate : candidates) {
-      if (!(candidate.hfflag() & 1 << DecayType::JpsiToEE)) {
+      if (d_selectionFlagJpsi > 0) {
+        if (!d_modeJpsiToMuMu) {
+          if (candidate.isSelJpsiToEE() <= 0)
+            continue;
+        } else {
+          if (candidate.isSelJpsiToMuMu() <= 0)
+            continue;
+        }
```

Comment:
This has a kind of reverted logic.
It makes more sense to do (and looks more transparent):
```c++
if (d_modeJpsiToMuMu) {
  if (candidate.isSelJpsiToMuMu() <= 0)
...
```